### PR TITLE
netsuite e2e setup and first test commit

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: cypress/e2e/journey/netsuite-journey.cy.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+describe('empty spec', () => {
+  it('passes', () => {
+    cy.visit('https://example.cypress.io')
```
Comment (ashwin1111) [inline cypress/e2e/journey/netsuite-journey.cy.ts:3]: remove / comment this out

File: cypress/e2e/micro_actions/auth/logout.cy.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+describe('logout', () => {
+  beforeEach(() => {
+    cy.ignoreTokenHealth()
+    cy.visit('/')
+  })
+
+  it('logout the user', () => {
+    cy.visit('/auth/logout')
+    cy.url().should('include', '/auth/login')
+    cy.get('[data-cy="ns-login"]').contains('Fyle NetSuite Integration')
```
Comment (ashwin1111) [inline cypress/e2e/micro_actions/auth/logout.cy.js:12]: create a custom command for querying the data attribute, can check partner dashboard

File: cypress/e2e/micro_actions/auth/logout.cy.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/// <reference types="cypress" />
+
+describe('logout', () => {
+  beforeEach(() => {
+    cy.ignoreTokenHealth()
+    cy.visit('/')
+  })
+
+  it('logout the user', () => {
+    cy.visit('/auth/logout')
```
Comment (ashwin1111) [inline cypress/e2e/micro_actions/auth/logout.cy.js:10]: implementation is mismatched with the doc
Screenshot 2022-12-12 at 3 58 58 PM

File: cypress/fixtures/example.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
```
Comment (ashwin1111) [inline cypress/fixtures/example.json:2]: not needed, can remove it off

File: cypress/support/commands.ts [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        saveSetting(content: string): void;
+        getMatToggle(toggleIndex: number): Cypress.Chainable<JQuery<HTMLElement>>;
+        ignoreTokenHealth(): void;
+        setupHttpListeners(): void;
+        navigateToSettingPage(pageName: string): void;
+        exportsPolling(): void;
+        waitForDashboardLoad(): void;
+        interrupt(): void;
+        navigateToModule(pageName: string): void;
+        navigateToMappingPage(pageName: string): void;
```
Comment (ashwin1111) [inline cypress/support/commands.ts:20]: few things would make sense only for new design apps. Please to remove all unrelated ones

File: cypress/support/commands.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  
+    setupInterceptor('GET', '/fyle/exportable_expense_groups', 'exportableExpenseGroups');
+  
+    setupInterceptor('GET', '/tasks/all/', 'tasksPolling');
+  
+    setupInterceptor('POST', '/exports/trigger', 'exportsTrigger');
+  
+    setupInterceptor('POST', '/mappings/employee', 'postEmployeeMapping');
+  
+    setupInterceptor('GET', '/errors/', 'getErrors');
```
Comment (ashwin1111) [inline cypress/support/commands.ts:86]: few things would make sense only for new design apps. Please to remove all unrelated ones

File: package.json [hunk 7]
```diff
@@ -7,7 +7,13 @@
     "build": "ng build",
     "test": "ng test",
     "lint": "ng lint",
-    "e2e": "ng e2e"
+    "e2e": "ng e2e",
+    "cypress:open": "cypress open",
+    "cypress_open": "cypress open -b chrome --e2e --env coverage=false",
```
Comment (ashwin1111) [inline package.json:12]: `cypress:open`, can remove 11th line

File: src/app/app.module.ts [hunk 8]
```diff
@@ -12,7 +12,7 @@ import { CoreModule } from './core/core.module';
 import { SharedModule } from './shared/shared.module';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 import { MatSnackBarModule, MAT_SNACK_BAR_DEFAULT_OPTIONS } from '@angular/material/snack-bar';
-import * as Sentry from '@sentry/angular';
+// import * as Sentry from '@sentry/angular';
```
Comment (ashwin1111) [inline src/app/app.module.ts:15]: remove all these lines

File: src/main.ts [hunk 9]
```diff
@@ -3,26 +3,26 @@ import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
...
-    ignoreErrors: [
-      'Non-Error exception captured'
-    ],
-    integrations: [new TracingIntegrations.BrowserTracing({
-      routingInstrumentation: Sentry.routingInstrumentation,
-    })],
-    tracesSampleRate: 0.1
-  });
-}
+// if (environment.sentry_dsn) {
```
Comment (ashwin1111) [inline src/main.ts:12]: remove

File: cypress/support/commands.ts (hunk 5 above)
Comment (DhaaraniCIT) [reply cypress/support/commands.ts:20]: will remove them while working

File: cypress/support/commands.ts (hunk 6 above)
Comment (DhaaraniCIT) [reply cypress/support/commands.ts:86]: will remove them while working

File: cypress/e2e/micro_actions/auth/logout.cy.js (hunk 3 above)
Comment (ashwin1111) [reply cypress/e2e/micro_actions/auth/logout.cy.js:10]: please login and logout from the dropdown. No one will use the URL or knows the URL. We'll need to write tests to simulate user actions.

File: package.json (hunk 7 above)
Comment (ashwin1111) [reply package.json:11]: `cypress:open` instead of `cypress_open`

File: cypress/support/commands.ts [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+  Cypress.Commands.add('waitForDashboardLoad', () => {
+    cy.wait('@synchronousImport').its('response.statusCode').should('equal', 200)
+    cy.wait('@exportableExpenseGroups').its('response.statusCode').should('equal', 200)  
+  })
+  
+  Cypress.Commands.add('navigateToModule', (pageName: string) => {
+    cy.getElement('side-nav').find('.netsuite-sidenav--nav-item').contains(pageName).click()
+  })
+  
+  Cypress.Commands.add('importToFyle', (fieldOrder: number, enable: boolean, optionName: string = '') => {
```
Comment (ashwin1111) [inline cypress/support/commands.ts:169]: same

File: cypress/support/commands.ts [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+    if (enable) {
+      cy.selectMatOption(optionName)
+    }
+  })
+  
+  Cypress.Commands.add('enableConfigurationToggle', (fieldOrder: number) => {
+    cy.getMatToggle(fieldOrder).click()
+  })
+  
+  Cypress.Commands.add('selectConfigurationField', (fieldOrder: number, optionName: string) => {
```
Comment (ashwin1111) [inline cypress/support/commands.ts:185]: same

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: cypress/support/commands.ts [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+      }
+    })
+  })
+  
+  Cypress.Commands.add('waitForDashboardLoad', () => {
+    cy.wait('@synchronousImport').its('response.statusCode').should('equal', 200)
+    cy.wait('@exportableExpenseGroups').its('response.statusCode').should('equal', 200)  
+  })
+  
+  Cypress.Commands.add('navigateToModule', (pageName: string) => {
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply cypress/support/commands.ts:165]: it is used in header e2e test

File: cypress/e2e/micro_actions/auth/auth.cy.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
+/// <reference types="cypress" />
+
+describe('auth module', () => {
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply cypress/e2e/micro_actions/auth/auth.cy.js:3]: added to the documentation

**Pull request merged**